packages/ra-ui-materialui/src/input/SelectInput.tsx

```diff
@@ -178,14 +178,14 @@ export const SelectInput = (props: SelectInputProps) => {
     ]);
 
     const handleChange = useCallback(
-        async (event: React.ChangeEvent<HTMLSelectElement>, newItem) => {
-            if (newItem) {
-                const value = getChoiceValue(newItem);
-                input.onChange(value);
-                return;
+        async (event: any) => {
+            let choice = event;
+            // eslint-disable-next-line eqeqeq
+            if (event?.target?.value != undefined) {
```

Comment:
what that code does is cryptic to me. Would you mind adding a comment explaining it?